### PR TITLE
Split storage AccountKey from storage account string

### PR DESCRIPTION
**Pull request by v-nagarajku**
**Task name**: AzureCloudPowerShellDeploymentV1

**Description**: This PR is to remove storage AccountKey string value from the storage account string.

**Documentation changes required:** N

**Added unit tests:** N

**Attached related issue:** https://portal.microsofticm.com/imp/v3/incidents/details/398837514/home

**Checklist**:
- [X ] Task version was bumped - please check [instruction](https://github.com/microsoft/azure-pipelines-tasks/tree/master/docs/taskversionbumping.md) how to do it
- [ X] Checked that applied changes work as expected

### CODE REVIEW

File: Tasks/AzureCloudPowerShellDeploymentV1/Utility.ps1 [hunk 1]
```diff
@@ -77,6 +77,19 @@ function Get-AzureStoragePrimaryKey($storageAccount, [bool]$isArm)
...
+{
+    if(-not ([string]::IsNullOrEmpty($publicConfigStorageAccountName)) -and $publicConfigStorageAccountName.Contains("AccountKey"))
+    {
+        $publicConfig_StorageAccountName = $publicConfigStorageAccountName -split ";"
+        return $publicConfig_StorageAccountName[0]
+    }
+    else{
+        return $publicConfigStorageAccountName
+    }
+
```
Comment (starkmsu) [inline Tasks/AzureCloudPowerShellDeploymentV1/Utility.ps1:90]: Why add an empty line?

File: Tasks/AzureCloudPowerShellDeploymentV1/Utility.ps1 [hunk 2]
```diff
@@ -77,6 +77,19 @@ function Get-AzureStoragePrimaryKey($storageAccount, [bool]$isArm)
     return $primaryStorageKey
 }
 
+function Get-SplitConfigStorageAccountName($publicConfigStorageAccountName)
+{
+    if(-not ([string]::IsNullOrEmpty($publicConfigStorageAccountName)) -and $publicConfigStorageAccountName.Contains("AccountKey"))
+    {
+        $publicConfig_StorageAccountName = $publicConfigStorageAccountName -split ";"
+        return $publicConfig_StorageAccountName[0]
```
Comment (starkmsu) [inline Tasks/AzureCloudPowerShellDeploymentV1/Utility.ps1:85]: I think this is not enough to extract account name from connection string like this - DefaultEndpointsProtocol=https;AccountName=storagesample;AccountKey=<account-key>

File: Tasks/AzureCloudPowerShellDeploymentV1/Utility.ps1 [hunk 3]
```diff
@@ -134,6 +148,7 @@ function Get-DiagnosticsExtensions($storageAccount, $extensionsPath, $storageAcc
                             try
                             {
                                 $publicConfigStorageKey = Get-AzureStoragePrimaryKey $publicConfigStorageAccountName $useArmStorage.IsPresent
+
```
Comment (starkmsu) [inline Tasks/AzureCloudPowerShellDeploymentV1/Utility.ps1:151]: Why add an empty line?

File: Tasks/AzureCloudPowerShellDeploymentV1/Utility.ps1 (hunk 1 above)
Comment (v-nagarajku) [reply Tasks/AzureCloudPowerShellDeploymentV1/Utility.ps1:90]: Will remove that empty line.

File: Tasks/AzureCloudPowerShellDeploymentV1/Utility.ps1 (hunk 2 above)
Comment (v-nagarajku) [reply Tasks/AzureCloudPowerShellDeploymentV1/Utility.ps1:85]: Our assumption from the Kusto logs that the string would be something like this 'hseydev;AccountKey=****'. To remove the AccountKey the above logic was used.

File: Tasks/AzureCloudPowerShellDeploymentV1/Utility.ps1 (hunk 3 above)
Comment (v-nagarajku) [reply Tasks/AzureCloudPowerShellDeploymentV1/Utility.ps1:151]: This empty line existed before code changes.

Comment (v-nagarajku) [reply Tasks/AzureCloudPowerShellDeploymentV1/Utility.ps1]: Removed it now and updated the PR. Thanks

**Review by starkmsu — APPROVED** [review]

**Pull request merged**